scripts/govDocES.py

```diff
@@ -27,20 +28,22 @@ def main():
 
     dbManager.createSession()
 
-    for work in dbManager.session.query(Work) \
+    batchSize = 1000
+    for work in dbManager.session.query(Work.uuid, array_agg(Edition.measurements)) \
         .join(Edition) \
         .filter(Edition.measurements != None) \
         .filter(Edition.measurements != []) \
-        .filter(Edition.measurements != [{}]).all():
+        .filter(Edition.measurements != [{}]) \
+        .group_by(Work.uuid) \
+        .yield_per(batchSize):
 
             break_out_flag = False
-            
-            for edition in work.editions:
-                for measurement in edition.measurements:
+            for editionMeasurements in work[0]:
+                for measurement in editionMeasurements:
                     if measurement['type'] == "government_document":
                         if measurement['value'] == "1":
                             try:
-                                workRec = ESWork.get(work.uuid, index=esManager.index)
+                                workRec = ESWork.get(work[0], index=esManager.index)
```

Comment:
Why is work.uuid replaced with work[0] here? Does it work better to find the exact work we're looking for?

Reply:
You already answered this in my previous question.